DQM/GEM/src/GEMDQMBase.cc

```diff
@@ -224,3 +237,124 @@ int GEMDQMBase::GenerateMEPerChamber(DQMStore::IBooker& ibooker) {
+    }
+
+    std::sort(listRadiusLower.begin(), listRadiusLower.end());
+    std::sort(listRadiusUpper.begin(), listRadiusUpper.end());
+
+    std::vector<Float_t>& listR =
+        (chamberNo % 2 == 0 ? stationInfo.listRadiusEvenChamber_ : stationInfo.listRadiusOddChamber_);
+    listR.clear();
+    listR.push_back(listRadiusLower.front());
+    for (int i = 1; i < (int)listRadiusLower.size(); i++) {
+      listR.push_back(0.5 * (listRadiusLower[i] + listRadiusUpper[i - 1]));
+    }
+    listR.push_back(listRadiusUpper.back());
+
+    if (chamberNo % 2 == 0)
+      bDoneEven = true;
+    if (chamberNo % 2 != 0)
+      bDoneOdd = true;
+
+    if (bDoneEven && bDoneOdd)
+      break;
+  }
+
+  return 0;
+}
+
+int GEMDQMBase::readGeometryPhiInfoChamber(const GEMStation* station, MEStationInfo& stationInfo) {
+  auto listSuperChambers = station->superChambers();
+  //Int_t nNumStripVFAT = stationInfo.nNumDigi_ / ( stationInfo.nMaxVFAT_ / stationInfo.nNumEtaPartitions_ );
+  Int_t nNumStripEta = stationInfo.nNumDigi_ * (stationInfo.nMaxVFAT_ / stationInfo.nNumEtaPartitions_);
+
+  std::vector<std::pair<Int_t, std::pair<std::pair<Float_t, Float_t>, Bool_t>>> listDivPhi;
+
+  // Obtaining phi intervals of chambers
+  for (auto superchamber : listSuperChambers) {
+    auto iEta = superchamber->chambers().front()->etaPartitions().front();
+
+    // What is the index of the first strip? Rather than to ask to someone, let's calculate it!
+    Float_t fWidthStrip = std::abs(iEta->centreOfStrip((Int_t)1).x() - iEta->centreOfStrip((Int_t)0).x());
+    LocalPoint lpRef(-fWidthStrip / 3.0, 0.0);
+    Int_t nStripMid = (Int_t)iEta->strip(lpRef);
+    Int_t nFirstStrip = 1 - ((nNumStripEta / 2) - nStripMid);
+    Int_t nLastStrip = nFirstStrip + nNumStripEta - 1;
+
+    auto& surface = iEta->surface();
+    LocalPoint lpF = iEta->centreOfStrip((Float_t)(nFirstStrip - 0.5));  // To avoid the round error(?)
+    LocalPoint lpL = iEta->centreOfStrip((Float_t)(nLastStrip + 0.5));   // To avoid the round error(?)
+    GlobalPoint gpF = surface.toGlobal(lpF);
+    GlobalPoint gpL = surface.toGlobal(lpL);
+
+    Float_t fPhiF = gpF.phi();
+    Float_t fPhiL = gpL.phi();
+    if (fPhiF * fPhiL < 0 && std::abs(fPhiF) > 0.5 * 3.14159265359) {
+      if (fPhiF < 0)
+        fPhiF += 2 * 3.14159265359;
+      if (fPhiL < 0)
+        fPhiL += 2 * 3.14159265359;
+    }
+    Bool_t bFlipped = fPhiF > fPhiL;
+    Float_t fPhiMin = std::min(fPhiF, fPhiL);
+    Float_t fPhiMax = std::max(fPhiF, fPhiL);
+
+    listDivPhi.emplace_back();
+    listDivPhi.back().first = iEta->id().chamber();
+    listDivPhi.back().second.first.first = fPhiMin;
+    listDivPhi.back().second.first.second = fPhiMax;
+    listDivPhi.back().second.second = bFlipped;
+  }
+
+  stationInfo.fMinPhi_ = 1048576.0;
+  for (auto p : listDivPhi) {
+    if (p.first == 1) {
+      stationInfo.fMinPhi_ = p.second.first.first;
+      break;
+    }
+  }
+
+  // For a future usage
```

Comment:
No sense keeping these commented lines

Reply:
> No sense keeping these commented lines

@quark2 Please address above comments.

Reply:
Hi @civanch and @qliphy, 

These lines are kept for future extensions. It's in printing-out format but if interfaces for this information are prepared, then these lines will be revived within the interface. This is the purpose of keeping these lines.